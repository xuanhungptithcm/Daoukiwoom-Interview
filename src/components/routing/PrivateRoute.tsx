import { ComponentType, createElement } from "react";
import { Redirect, Route } from "react-router-dom";
import { AUTH_PATH } from "./constants";

interface IContentRouteProps {
  path?: string;
  exact?: boolean;
  isLogged?: boolean;
  component?: ComponentType;
}
const PrivateRoute = ({
  component,
  exact,
  isLogged,
  ...props
}: IContentRouteProps) => {
  return (
    <Route exact={exact} {...props}>
      {(routeProps) => {
        if (!routeProps.match) return null;
        if (!isLogged) return <Redirect to={AUTH_PATH.LOGIN_PAGE} />;
        if (component) {
          return <>{createElement(component)}</>;
        }
        return null;
      }}
    </Route>
  );
};

export default PrivateRoute;
