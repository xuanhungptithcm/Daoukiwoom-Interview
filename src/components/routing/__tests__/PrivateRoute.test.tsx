/**
 * @jest-environment jsdom
 */

import { shallow } from "enzyme";
import PrivateRoute from "../PrivateRoute";

describe("PrivateRoute", () => {
  test("should properly render with authorized", () => {
    const data: any = {
      component: "A",
      exact: true,
      isAuthorized: true,
      path: "/a",
    };
    expect(shallow(<PrivateRoute {...data} />)).toMatchSnapshot();
  });
});
