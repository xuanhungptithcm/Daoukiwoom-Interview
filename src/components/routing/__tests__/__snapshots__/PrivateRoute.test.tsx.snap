// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute should properly render with authorized 1`] = `
<Route
  exact={true}
  path="/a"
  render={[Function]}
/>
`;
